airbyte-workers/src/main/java/io/airbyte/workers/WorkerConfigs.java

```diff
@@ -22,22 +25,53 @@
   private final String jobBusyboxImage;
   private final String jobCurlImage;
   private final Map<String, String> envMap;
+  private final Duration workerStatusCheckInterval;
 
+  private static final Duration DEFAULT_WORKER_STATUS_CHECK_INTERVAL = Duration.ofSeconds(30);
+  private static final Duration CHECK_WORKER_STATUS_CHECK_INTERVAL = Duration.ofSeconds(1);
+
+  /**
+   * Constructs a job-type-agnostic WorkerConfigs. For WorkerConfigs customized for specific
+   * job-types, use static `build*JOBTYPE*WorkerConfigs` method if one exists.
+   */
   public WorkerConfigs(final Configs configs) {
-    this.workerEnvironment = configs.getWorkerEnvironment();
-    this.resourceRequirements = new ResourceRequirements()
-        .withCpuRequest(configs.getJobMainContainerCpuRequest())
-        .withCpuLimit(configs.getJobMainContainerCpuLimit())
-        .withMemoryRequest(configs.getJobMainContainerMemoryRequest())
-        .withMemoryLimit(configs.getJobMainContainerMemoryLimit());
-    this.workerKubeTolerations = configs.getJobKubeTolerations();
-    this.workerKubeNodeSelectors = configs.getJobKubeNodeSelectors();
-    this.jobImagePullSecret = configs.getJobKubeMainContainerImagePullSecret();
-    this.jobImagePullPolicy = configs.getJobKubeMainContainerImagePullPolicy();
-    this.jobSocatImage = configs.getJobKubeSocatImage();
-    this.jobBusyboxImage = configs.getJobKubeBusyboxImage();
-    this.jobCurlImage = configs.getJobKubeCurlImage();
-    this.envMap = configs.getJobDefaultEnvMap();
+    this(configs.getWorkerEnvironment(),
+        new ResourceRequirements()
+            .withCpuRequest(configs.getJobMainContainerCpuRequest())
+            .withCpuLimit(configs.getJobMainContainerCpuLimit())
+            .withMemoryRequest(configs.getJobMainContainerMemoryRequest())
+            .withMemoryLimit(configs.getJobMainContainerMemoryLimit()),
+        configs.getJobKubeTolerations(),
+        configs.getJobKubeNodeSelectors(),
+        configs.getJobKubeMainContainerImagePullSecret(),
+        configs.getJobKubeMainContainerImagePullPolicy(),
+        configs.getJobKubeSocatImage(),
+        configs.getJobKubeBusyboxImage(),
+        configs.getJobKubeCurlImage(),
+        configs.getJobDefaultEnvMap(),
+        DEFAULT_WORKER_STATUS_CHECK_INTERVAL);
+  }
+
+  /**
+   * Builds a WorkerConfigs with some configs that are specific to the Check job type.
+   */
+  public static WorkerConfigs buildCheckWorkerConfigs(final Configs configs) {
+    return new WorkerConfigs(
+        configs.getWorkerEnvironment(),
+        new ResourceRequirements()
+            .withCpuRequest(configs.getJobMainContainerCpuRequest())
+            .withCpuLimit(configs.getJobMainContainerCpuLimit())
+            .withMemoryRequest(configs.getJobMainContainerMemoryRequest())
+            .withMemoryLimit(configs.getJobMainContainerMemoryLimit()),
+        configs.getJobKubeTolerations(),
+        configs.getCheckJobKubeNodeSelectors(),
```

Comment:
for the fallback behavior should this be?
```suggestion
        configs.getCheckJobKubeNodeSelectors() != null? configs.getCheckJobKubeNodeSelectors() : configs.getJobKubeNodeSelectors(),
```